components/update_client/BUILD.gn

```diff
@@ -247,7 +247,7 @@ target(gtest_target_type, "cobalt_slot_management_test") {
     "//testing/gmock",
     "//testing/gtest",
   ]
-  if (build_with_separate_cobalt_toolchain) {
+  if (is_modular_toolchain) {
```

Comment:
This check looks suspicious, whey do we even need it?
Only Evergreen should be building the udpate_client.

Reply:
Makes sense , ill add sb_is_evergreen here

 alongwith other comments you may have in pass2

Reply:
DOne